src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -45,6 +45,21 @@ public CommandResult execute(Model model) throws CommandException {
         return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, personToDelete));
     }
 
+    @Override
+    public CommandResult executeBatch(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
+        model.deletePerson(personToDelete);
+
+        return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, personToDelete));
+    }
+
```

Comment:
What is the reason we need this method, since it is basically the same as the one above?

Reply:
This method does not revert the model's filtered list to the unfiltered list. Else, there will be indexing error if the index in the filtered list (which is what the user passed in e.g. in the event of find then batch) is not equals to the index in the unfiltered list, as the first `execute()` will return the list to the unfiltered list, which the subsequent batch commands will execute on.

The updating is only done in the `BatchCommand's` execute method once all the `executebatch()` are completed.